### DISCUSSION TITLE
Scoped Containers - Best DI

### DISCUSSION THREAD

**Discussion by tzarger** [MichalLytek/type-graphql#1227] [Q&A] [answered]
I am starting to use scoped containers and I saw a note about:

> Be aware that some libraries (like TypeDI) by default create new instances for every scoped container, so you might experience a significant increase in memory usage and some slowing down in query resolving speed, so please be careful with using this feature!

Is there a better DI to use than TypeDI (which I enjoy using TypeDI) in regard to performance when using scoped containers?

Thanks!

**Comment by MichalLytek** [comment, marked as answer]
I would use tsyringe now

**Reply by tzarger** [reply]
@MichalLytek when trying to assign the container in buldSchema:

```
export const schema = buildSchemaSync({
  resolvers,
  container: container,
  // container: ({ context }: ResolverData<GraphQLContext>) => context.container,
});
```
I received the following error:

`Type 'DependencyContainer' is not assignable to type 'ContainerType | ContainerGetter<any> | undefined'.ts(2322)`

I looked on the `container` and I do not see anything that would give me the correct type. On another note, I like how simple tsyringe is!

**Reply by MichalLytek** [reply]
`ContainerType` need `.get()` method while `tsyringe` gives you `.resolve()`

**Reply by tzarger** [reply]
@MichalLytek I assume with tsyringe you cannot use the container in build schema, correct? You would need to import wherever you want, like a traditional DI scenario...

**Reply by MichalLytek** [reply]
```ts
container: {
  get: (cls) => container.resolve(cls),
}
```
It's really that simple 😉

**Reply by tzarger** [reply]
Thanks @MichalLytek worked like a charm! Much appreciate your time and suggestions.
